client/state/posts/utils.js

```diff
@@ -21,22 +21,31 @@ export function getNormalizedPostsQuery( query ) {
 
 /**
  * Returns a serialized posts query, used as the key in the
- * `state.posts.siteQueries` state object.
+ * `state.posts.queries` state object.
  *
- * @param  {Object} query Posts query
- * @return {String}       Serialized posts query
+ * @param  {Object} query  Posts query
+ * @param  {Number} siteId Optional site ID
+ * @return {String}        Serialized posts query
  */
-export function getSerializedPostsQuery( query = {} ) {
-	return JSON.stringify( getNormalizedPostsQuery( query ) ).toLowerCase();
+export function getSerializedPostsQuery( query = {}, siteId ) {
```

Comment:
@retrofox added [jshashes](https://github.com/h2non/jshashes) as a dependency for the sync handler, so md5 is available if you'd like to use it.